### PR TITLE
#910: [typescript .d.ts] DropdownMenu (closes #910)

### LINKED ISSUE #910: [typescript .d.ts] DropdownMenu
← #908

## requirements
{describe the new feature}

## specs
{optional: describe technical specs to implement this feature, if not obvious}

## misc
{optional: other useful info}

### PR DESCRIPTION
**Pull request by ascariandrea**
Closes #910

## Test Plan

> These results can be screenshots, query results, or even just pointers to unit tests that successfully passed.

- [ ] Google Chrome
- [ ] Internet Explorer

> An example of partial test is a field displaying 4 possible values. If 3 values are tested, with screenshots, and 1 is not, then it should be mentioned here.

### CODE REVIEW

**Comment by FrancescoCioria** [comment]
@ascariandrea can you rebase? (there are 17 changed files in this PR but I suppose most of them are already like this on `master`)

**Comment by ascariandrea** [comment]
@FrancescoCioria done!

File: src/DropdownMenu/DropdownMenu.d.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { PureComponet } from 'react';
+import { MenuProps } from './Menu';
+
+export type DropdownMenuProps = {
+  children?: any, // TODO: t.ReactChildren
+  menuRenderer?: (options: MenuProps.Option) => JSX.Element,
```
Comment (FrancescoCioria) [inline src/DropdownMenu/DropdownMenu.d.ts]: this is a list of options: `MenuProps.Option[]`

File: src/DropdownMenu/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import DropdownMenu from './DropdownMenu';
```
Comment (FrancescoCioria) [inline src/DropdownMenu/index.d.ts]: minor: If I'm not mistaken, when using only `export` you can do `export X from './X` (it doesn't work only for `export default`)

File: src/DropdownMenu/DropdownMenu.d.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { PureComponet } from 'react';
```
Comment (FrancescoCioria) [inline src/DropdownMenu/DropdownMenu.d.ts]: typo! 🚨

**Pull request merged**